### PR TITLE
Remove security hotspots, Fix #285

### LINKED ISSUE #285: Remove security hotspots

### PR DESCRIPTION
**Pull request by minborg**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/core/util/ThreadIndexAssigner.java [hunk 1]
```diff
@@ -44,6 +44,6 @@ public int getId() throws IllegalStateException, BufferOverflowException, Buffer
     }
 
     protected int nextIndex(int size) {
-        return ThreadLocalRandom.current().nextInt(size);
+        return new SecureRandom().nextInt(size);
```
Comment (nicktindall) [inline src/main/java/net/openhft/chronicle/core/util/ThreadIndexAssigner.java:47]: Is it worth implementing a `ThreadLocalSecureRandom` (similar to `ThreadLocalRandom`), this seems like a sonar warning that will come up again and again? Also Core might be a good place to put it.

Or does something like that already exist in the standard library?

**Review by nicktindall — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/core/Jvm.java [hunk 2]
```diff
@@ -1587,7 +1587,7 @@ public static String getCpuClass() {
                 if (Files.isReadable(path)) {
                     model = Files.lines(path)
                             .filter(line -> line.startsWith("model name"))
-                            .map(line -> line.replaceAll(".*: ", ""))
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/core/Jvm.java:1590]: does this code have tests?

File: src/main/java/net/openhft/chronicle/core/util/ThreadIndexAssigner.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/core/util/ThreadIndexAssigner.java:47]: Or shall we just suppress the sonar warning? I don't see it provides much benefit - we don't use randoms for anything important, or attackable

Comment (minborg) [reply src/main/java/net/openhft/chronicle/core/util/ThreadIndexAssigner.java:47]: In this case, there is no performance issue so I think we can stick with SecureRandom

File: src/main/java/net/openhft/chronicle/core/Jvm.java (hunk 2 above)
Comment (minborg) [reply src/main/java/net/openhft/chronicle/core/Jvm.java:1590]: No. We should break out this functionality into a separate method and create a test.

Comment (minborg) [reply src/main/java/net/openhft/chronicle/core/Jvm.java:1590]: Added a test

**Pull request merged**